src/inner_product_proof.rs

```diff
@@ -51,6 +51,15 @@ impl InnerProductProof {
 
         let mut n = G.len();
 
+        if n == 0 {
+            return InnerProductProof {
+                L_vec: vec![],
+                R_vec: vec![],
+                a: Scalar::zero(),
+                b: Scalar::zero(),
+            };
+        }
```

Comment:
What is the meaning of an `InnerProductProof` about empty vectors? Is this just to cover the case where a circuit has no multiplication gates?

Reply:
Great. We should probably double-check the behaviour of the function and write a note about what inputs are allowed (what length of vectors, whether they have to be powers of two, etc).

Reply:
Maybe we can come up with a reason why this function is OK even when the inputs aren't power-of-two lengths.